_posts/2021-07-25-personal-allowance.md

```diff
@@ -0,0 +1,1319 @@
+    });
+  </script>
+</div>
+<div id = "graph_graph_28_1"></div>
+
+On average, the policy would serve as a transfer from older adults to children. The average child under 18 would see their household net income rise by £2,622 under full replacement of the personal allowance with a UBI.  
+
+
+<button onclick="show_code_30()">Click to show code</button>
+<div id="code_block_30" style="display: none;">
+  <pre>
+    <code>
+#Mean difference in household income by age (calc)
+ages = []
+for i in range(81):
+    ages.append(i)
+ages = pd.Series(ages)
+
+def income_by_age(sim_BI):
+    gains_by_age = []
+    x1 = sim_BI.calc("household_net_income", map_to="person") - sim.calc("household_net_income", map_to="person")
+    for n in range(81):
+        values = x1[age.round() == n]
+        mean = values.sum()/values.count()
+        gains_by_age.append(mean)
+    return gains_by_age
+    </code>
+  </pre>
+</div>
+
+<script>
+function show_code_30() {
+  var x = document.getElementById("code_block_30");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+</script>
+
+
+<button onclick="show_code_31()">Click to show code</button>
+<div id="code_block_31" style="display: none;">
+  <pre>
+    <code>
+#Mean difference in household income by age (graph)
+df_list = []
+sim_num = 0
+age = sim.calc("age", map_to ="person")
+for i in PA_amounts2:
+    df = pd.DataFrame({"Age": ages, "Income change": income_by_age(sim_list[sim_num]), "PA amount": i})
+    df_list.append(df)
+    sim_num += 1
+    
+final_df = pd.concat(df_list)
+final_df
+format_fig(px.bar(final_df, x="Age", y="Income change", animation_frame="PA amount", range_y=[-1100,3100],
+                 hover_data={"PA amount": False}).update_layout(
+    title_text='Average change in household net income by age',
+    yaxis_title="Change in household net income",
+    yaxis_tickprefix="£",
+    xaxis_title="Age",
+    showlegend= False,
+    ).update_traces(marker_color='#1976D2'))
+    </code>
+  </pre>
+</div>
+
+<script>
+function show_code_31() {
+  var x = document.getElementById("code_block_31");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+</script>
+
+<div>
+  <script>
+    $(document).ready(function(){
+      $("#graph_graph_31_1").load("{{site.baseurl}}assets/markdown_assets/personal-allowance/graph_31_1.html");
+    });
+  </script>
+</div>
+<div id = "graph_graph_31_1"></div>
+
+#### Impacts for specific household types
+Overall, 52% of people come out ahead, most of them belonging to the bottom four deciles. 91% of the lowest decile, 73% of the second-lowest decile, and 62% of the third-lowest decile would see their incomes rise by over 5%. In contrast, most people who see their income decrease are in the upper deciles and lose less than 5%.
```

Comment:
just a note that using the slider on this chart creates an awesome visual effect, love it